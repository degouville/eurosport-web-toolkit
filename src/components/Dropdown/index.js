import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import Chevron from 'src/assets/chevron.component.svg';
import { medium } from '../../breakpoints';
import { H3 } from '../../typography';

export const StyledContainer = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
`;

export const StyledOverlay = styled.div`
  position: fixed;
  width: 100vw;
  height: 100vh;
  top: 0;
  left: 0;
  z-index: 1;

  display: none;
  ${({ isDropdownOpen }) =>
    isDropdownOpen &&
    css`
      display: block;
    `}
`;

export const StyledWrapper = styled.div`
  ${medium(css`
    width: fit-content;
  `)};
  position: relative;
`;

export const StyledText = styled(H3)`
  color: ${({ theme }) => theme.dropdown.label.color};
  text-transform: uppercase;
`;

export const StyledArrow = styled(Chevron)`
  height: 12px;
  width: 12px;
  transform: rotate(90deg);
  margin-left: 5px;
  path {
    fill: ${({ theme }) => theme.dropdown.label.color};
  }
`;

export const StyledDropdown = styled.div`
  position: absolute;
  width: calc(100% - 25px);
  top: 45px;
  ${medium(css`
    right: -94px;
    @-moz-document url-prefix() {
      right: -91px;
    }
    @supports (-ms-ime-align: auto) {
      right: unset;
    }
    bottom: auto;
    width: fit-content;
    max-width: 183px;
  `)};
  display: none;
  z-index: 2;
  opacity: 0.95;
  align-self: flex-end;
  border: solid 1px ${({ theme }) => theme.dropdown.border};
  border-radius: 3px;
  background-color: ${({ theme }) => theme.dropdown.background};
  padding: 17px 8px 12px 17px;
  ${({ isDropdownOpen }) =>
    isDropdownOpen &&
    css`
      display: block;
    `}
  &::before {
    content: '';
    position: absolute;
    width: 10px;
    border-top: 1px solid ${({ theme }) => theme.dropdown.border};
    border-left: 1px solid ${({ theme }) => theme.dropdown.border};
    border-radius: 1px;
    background: ${({ theme }) => theme.dropdown.background};
    height: 10px;
    display: block;
    margin-left: 50%;
    transform: translateX(-50%) rotate(45deg);
    top: -6px;
    left: 0;
  }
`;

const StyledUl = styled.ul`
  height: 130px;
  overflow-y: scroll;
  ::-webkit-scrollbar {
    width: 6px;
  }

  ::-webkit-scrollbar-track {
    background: ${({ theme }) => theme.scrollbar.background};
    border-radius: 2px;
  }

  ::-webkit-scrollbar-thumb {
    background: ${({ theme }) => theme.scrollbar.color};
    border-radius: 2px;
  }
`;

const StyledLi = styled.li`
  color: ${({ isSelected, theme }) =>
    isSelected ? theme.dropdown.list.selected.color : theme.dropdown.list.default.color};
  font-size: 14px;
  cursor: pointer;
  padding: 13px 0;
  margin-right: 10px;
  display: flex;
  justify-content: space-between;
  &:first-of-type {
    padding-top: 0;
  }
  &:not(:last-of-type) {
    border-bottom: solid 1px ${({ theme }) => theme.dropdown.border};
  }
  &:hover {
    color: ${({ theme }) => theme.dropdown.list.selected.color};
  }
`;

export const StyledCheckMark = styled.span`
  color: ${({ theme }) => theme.dropdown.checkMark.color};
  margin: 0 0 0 1em;
`;

const StyledP = styled.p`
  width: 100%;
  ${medium(css`
    width: 130px;
  `)};
`;

class Dropdown extends React.Component {
  state = {
    isDropdownOpen: false,
    selectedOption: null,
  };

  componentWillMount() {
    this.setSelectedOption();
  }

  setSelectedOption = () => {
    const { options } = this.props;
    this.setState({ selectedOption: options[0] });
  };

  onOptionClick = id => {
    const { options, onItemSelected } = this.props;
    const matchingOption = options.find(option => option.id === id);
    onItemSelected && onItemSelected(matchingOption);

    if (matchingOption) {
      this.setState({ selectedOption: matchingOption, isDropdownOpen: false });
    }
  };

  toggleOpen = () => {
    // eslint-disable-next-line no-unused-vars
    this.setState(state => ({
      isDropdownOpen: !state.isDropdownOpen,
    }));
  };

  render() {
    const { options } = this.props;
    const { isDropdownOpen, selectedOption } = this.state;
    return (
      <StyledWrapper>
        <StyledOverlay isDropdownOpen={isDropdownOpen} onClick={this.toggleOpen} />
        <StyledDropdown isDropdownOpen={isDropdownOpen}>
          <StyledUl>
            {options &&
              options.map(option => (
                <StyledLi
                  onClick={() => this.onOptionClick(option.id)}
                  isSelected={selectedOption.id === option.id}
                  key={option.id}
                >
                  <StyledP>{option.text}</StyledP>
                  {selectedOption.id === option.id && <StyledCheckMark> âœ“</StyledCheckMark>}
                </StyledLi>
              ))}
          </StyledUl>
        </StyledDropdown>
        <StyledContainer onClick={this.toggleOpen}>
          <StyledText>
            {selectedOption.text}
            <StyledArrow />
          </StyledText>
        </StyledContainer>
      </StyledWrapper>
    );
  }
}

Dropdown.defaultProps = {
  onItemSelected: null,
};

Dropdown.propTypes = {
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      text: PropTypes.string,
    })
  ).isRequired,
  onItemSelected: PropTypes.func,
};

export default Dropdown;
