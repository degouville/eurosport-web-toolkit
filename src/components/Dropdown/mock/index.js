export const dropdownOptions = [
  { text: 'Men singles', id: 1231 },
  { text: 'Women singles', id: 1232 },
  { text: 'Men doubles', id: 1233 },
  { text: 'Women doubles', id: 1234 },
  { text: 'Men triples', id: 1235 },
  { text: 'Women triples', id: 1236 },
];

export const initialOptionID = dropdownOptions[3].id;

export default {
  ...dropdownOptions,
  ...initialOptionID,
};
