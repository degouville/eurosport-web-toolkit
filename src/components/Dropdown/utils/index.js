// eslint-disable-next-line import/prefer-default-export
export const removeUnit = width => parseInt(width, 10);
