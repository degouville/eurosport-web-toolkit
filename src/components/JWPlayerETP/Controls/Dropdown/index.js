import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';

const Dropdown = ({ children, bottomDisplay }) => (
  <MainContainer>
    <StyledContainer bottomDisplay={bottomDisplay}>
      <StyledDropdownPick bottomDisplay={bottomDisplay} />
      <StyledDropdown>{children}</StyledDropdown>
    </StyledContainer>
  </MainContainer>
);

Dropdown.propTypes = {
  children: PropTypes.node.isRequired,
  bottomDisplay: PropTypes.bool.isRequired,
};

const MainContainer = styled.div`
  position: relative;
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
`;

const StyledContainer = styled.div`
  position: absolute;
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
  flex-direction: ${({ bottomDisplay }) => (bottomDisplay ? 'column-reverse' : 'column')};
  ${({ bottomDisplay }) => bottomDisplay && 'bottom: 100%;'}
`;

const StyledDropdownPick = styled.div`
  border: solid 1px ${({ theme }) => theme.playerControls.dropdown.border};
  ${({ bottomDisplay }) => bottomDisplay && 'border-left: none;'}
  ${({ bottomDisplay }) => bottomDisplay && 'border-top: none;'}
  ${({ bottomDisplay }) => !bottomDisplay && 'border-right: none;'}
  ${({ bottomDisplay }) => !bottomDisplay && 'border-bottom: none;'}
   
  border-radius: 3px;
  background-color: ${({ theme }) => theme.playerControls.dropdown.background};
  height: 10px;
  width: 10px;
  transform: ${({ bottomDisplay }) => (bottomDisplay ? ' translate(0%, -50%)' : 'translate(0%, 50%)')} rotate(45deg);
`;

export const StyledDropdown = styled.div`
  border: solid 1px ${({ theme }) => theme.playerControls.dropdown.border};
  border-radius: 3px;
  background-color: ${({ theme }) => theme.playerControls.dropdown.background};
  padding: 18px;
`;

export default Dropdown;
