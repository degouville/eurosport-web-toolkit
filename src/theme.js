import { transparentize, rgba } from 'polished';
import {
  plausibleViolet,
  crashingAmber,
  coreLightMinus1,
  significantAzure,
  featureOneMinus2,
  actionTwoDarkBase,
  coreDarkPlus1,
  coreNeutral2,
  coreNeutral4,
  coreNeutral8,
  coreLightBase,
  dodgerBlue2,
  actionTwoLightBase,
  melanzane,
  coreDarkBase,
  hydroelectricTar,
  coreNeutral6,
  coreNeutral1,
  actionOneDarkMinus1,
  actionOneDarkBase,
  ebonyClay,
  martinique,
  suitGray,
  blackPearl,
} from './colors';
import { fontAlphaHeadline, fontInterUi } from './typography';

export default {
  underlineLink: {
    fontFamily: fontInterUi,
    color: coreNeutral4,
    hoverColor: coreNeutral2,
  },
  link: {
    color: featureOneMinus2,
  },
  button: {
    primary: {
      color: coreLightMinus1,
      backgroundColor: actionTwoDarkBase,
    },
    secondary: {
      color: featureOneMinus2,
    },
    form: {
      color: coreLightMinus1,
      backgroundColor: dodgerBlue2,
    },
  },
  input: {
    backgroundColor: coreDarkPlus1,
    borderFocused: coreNeutral4,
    border: coreNeutral8,
    placeholder: {
      color: coreNeutral4,
      fontFamily: fontInterUi,
    },
    text: {
      color: coreLightBase,
      fontFamily: fontInterUi,
    },
  },
  login: {
    separator: {
      primary: coreLightBase,
      secondary: coreNeutral8,
    },
    text: {
      color: coreLightBase,
      fontFamily: fontInterUi,
    },
    title: {
      color: coreLightMinus1,
      fontFamily: fontAlphaHeadline,
    },
  },
  banner: {
    error: {
      borderColor: actionTwoLightBase,
      backgroundColor: melanzane,
      textColor: coreLightBase,
      fontFamily: fontInterUi,
    },
  },
  tab: {
    active: {
      color: coreLightMinus1,
      borderColor: featureOneMinus2,
    },
    borderColor: transparentize(0.5, coreLightMinus1),
  },
  typo: {
    primary: {
      color: coreLightMinus1,
    },
  },
  playerControls: {
    dropdown: {
      background: transparentize(0.1, coreDarkBase),
      border: transparentize(0.2, coreNeutral6),
    },
    seek: {
      trackColor: dodgerBlue2,
      handleSize: 20,
      handleColor: coreLightBase,
      railColor: rgba(coreNeutral1, 0.3),
      railThickness: 5,
      vertical: false,
    },
    volume: {
      trackColor: dodgerBlue2,
      handleSize: 20,
      handleColor: coreLightBase,
      railColor: rgba(coreNeutral1, 0.3),
      railThickness: 5,
      vertical: true,
    },
    bar: {
      backgroundColor: transparentize(0.4, coreDarkBase),
    },
    separator: transparentize(0.5, coreNeutral4),
    liveIcon: {
      fontFamily: fontAlphaHeadline,
      backgroundColor: actionTwoDarkBase,
      color: coreLightBase,
    },
    text: {
      fontFamily: fontAlphaHeadline,
      color: coreLightBase,
    },
  },
  dropdown: {
    label: {
      color: coreLightMinus1,
    },
    list: {
      default: {
        color: coreNeutral2,
      },
      selected: {
        color: coreLightMinus1,
      },
    },
    checkMark: {
      color: significantAzure,
    },
    background: hydroelectricTar,
    border: transparentize(0.2, coreNeutral6),
  },
  scrollbar: {
    color: plausibleViolet,
    background: crashingAmber,
  },
  statistics: {
    comparison: {
      colorOne: actionOneDarkMinus1,
      colorTwo: actionOneDarkBase,
      background: rgba(coreLightMinus1, 0.1),
    },
    background: {
      one: ebonyClay,
      two: martinique,
    },
    fringe: rgba(coreLightMinus1, 0.05),
  },
  marketingChecklist: {
    title: {
      color: coreLightMinus1,
    },
    subtitle: {
      color: suitGray,
    },
    check: {
      color: coreLightMinus1,
    },
  },
  videoPlayerModal: {
    closeIcon: {
      fill: coreLightMinus1,
    },
  },
  loginWithMarketing: {
    mobileBackground: blackPearl,
  },
};
