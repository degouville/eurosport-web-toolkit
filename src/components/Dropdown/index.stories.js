import React from 'react';
import { storiesOf } from '@storybook/react';
import { object, number } from '@storybook/addon-knobs';
import { Dropdown } from 'src';
import { dropdownOptions, initialOptionID } from './mock';

const dropdownStories = storiesOf('Components|Dropdown', module);

dropdownStories
  .add('Default', () => <Dropdown options={object('options', dropdownOptions)} />)
  .add('Default with 1 item', () => <Dropdown options={object('options', dropdownOptions.slice(0, 1))} />)
  .add('With Initial Option', () => (
    <Dropdown
      options={object('options', dropdownOptions)}
      initialOptionID={number('InitialOptionID', initialOptionID)}
    />
  ));
