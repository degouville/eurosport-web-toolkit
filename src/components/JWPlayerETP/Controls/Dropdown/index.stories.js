import React from 'react';
import { storiesOf } from '@storybook/react';
import theme from 'src/theme';
import { boolean } from '@storybook/addon-knobs';
import styled from 'react-emotion';
import Dropdown from './index';
import Slider from '../Slider';

const dropdownStories = storiesOf('Components|PlayerDropdown', module);

dropdownStories.add('Default', () => (
  <MainContainer>
    <Dropdown theme={theme.playerControls.dropdown} bottomDisplay={boolean('bottomDisplay', true)}>
      <SliderContainer>
        <Slider max={100} onChange={() => null} min={0} value={50} defaultValue={0} {...theme.playerControls.volume} />
      </SliderContainer>
    </Dropdown>
  </MainContainer>
));

const MainContainer = styled.div`
  margin-top: 200px;
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
`;

const SliderContainer = styled.div`
  height: 100px;
`;
