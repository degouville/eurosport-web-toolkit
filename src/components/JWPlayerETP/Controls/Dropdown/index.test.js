import React from 'react';
import { mount } from 'enzyme/build';
import { ThemeProvider } from 'emotion-theming';
import theme from 'src/theme';
import Dropdown from './index';

describe('Dropdown Component', () => {
  const Child = () => <div />;

  it('Should match snapshot with bottomDisplay true', () => {
    // Given
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Dropdown bottomDisplay>
          <Child />
        </Dropdown>
      </ThemeProvider>
    );

    // Expect
    expect(wrapper.find(Dropdown)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('Should match snapshot with bottomDisplay false', () => {
    // Given
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Dropdown bottomDisplay={false}>
          <Child />
        </Dropdown>
      </ThemeProvider>
    );

    // Expect
    expect(wrapper.find(Dropdown)).toMatchSnapshot();
    wrapper.unmount();
  });
});
