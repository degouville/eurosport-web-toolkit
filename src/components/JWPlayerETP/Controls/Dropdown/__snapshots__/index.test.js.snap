// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component Should match snapshot with bottomDisplay false 1`] = `
.emotion-6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border: solid 1px rgba(107,108,128,0.8);
  border-radius: 3px;
  background-color: rgba(21,23,38,0.9);
  padding: 18px;
}

.emotion-4 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  border: solid 1px rgba(107,108,128,0.8);
  border-right: none;
  border-bottom: none;
  border-radius: 3px;
  background-color: rgba(21,23,38,0.9);
  height: 10px;
  width: 10px;
  -webkit-transform: translate(0%,50%) rotate(45deg);
  -ms-transform: translate(0%,50%) rotate(45deg);
  transform: translate(0%,50%) rotate(45deg);
}

<Dropdown
  bottomDisplay={false}
>
  <Styled(div)>
    <div
      className="emotion-6 emotion-7"
    >
      <Styled(div)
        bottomDisplay={false}
      >
        <div
          className="emotion-4 emotion-5"
        >
          <Styled(div)
            bottomDisplay={false}
          >
            <div
              className="emotion-0 emotion-1"
            />
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-2 emotion-3"
            >
              <Child>
                <div />
              </Child>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </div>
  </Styled(div)>
</Dropdown>
`;

exports[`Dropdown Component Should match snapshot with bottomDisplay true 1`] = `
.emotion-6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  bottom: 100%;
}

.emotion-0 {
  border: solid 1px rgba(107,108,128,0.8);
  border-left: none;
  border-top: none;
  border-radius: 3px;
  background-color: rgba(21,23,38,0.9);
  height: 10px;
  width: 10px;
  -webkit-transform: translate(0%,-50%) rotate(45deg);
  -ms-transform: translate(0%,-50%) rotate(45deg);
  transform: translate(0%,-50%) rotate(45deg);
}

.emotion-2 {
  border: solid 1px rgba(107,108,128,0.8);
  border-radius: 3px;
  background-color: rgba(21,23,38,0.9);
  padding: 18px;
}

<Dropdown
  bottomDisplay={true}
>
  <Styled(div)>
    <div
      className="emotion-6 emotion-7"
    >
      <Styled(div)
        bottomDisplay={true}
      >
        <div
          className="emotion-4 emotion-5"
        >
          <Styled(div)
            bottomDisplay={true}
          >
            <div
              className="emotion-0 emotion-1"
            />
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-2 emotion-3"
            >
              <Child>
                <div />
              </Child>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </div>
  </Styled(div)>
</Dropdown>
`;
